lib/checkstl.cpp

```diff
@@ -1237,6 +1237,142 @@ void CheckStl::if_findError(const Token *tok, bool str)
         reportError(tok, Severity::warning, "stlIfFind", "Suspicious condition. The result of find() is an iterator, but it is not properly checked.", CWE398, false);
 }
 
+static std::pair<const Token *, const Token *> isMapFind(const Token *tok)
+{
+    if (!Token::simpleMatch(tok, "("))
+        return {};
+    if (!Token::simpleMatch(tok->astOperand1(), "."))
+        return {};
+    if (!astIsContainer(tok->astOperand1()->astOperand1()))
+        return {};
+    if (!Token::Match(tok->astOperand1(), ". find|count ("))
```

Comment:
I think the `Library::getAction` should be used here.

Reply:
How does that work?

Reply:
It can be used to see what container methods do.

Something like: `tok->astOperand1()->astOperand1()->valueType()->container->getAction("find")` should return `Library::Container::Action::FIND`.

Reply:
What about `count`?

Reply:
The `count` is not covered yet by library. Maybe we should add a `CONTAINS` yield. I think the name `isMapFind` is confusing. It is not checked anywhere if the container is a `std::map`, `QMap` or even some `QList` etc. I assume this function will return stuff even for some `QList::find()` call.

Reply:
> It is not checked anywhere if the container is a std::map, QMap or even some QList etc. I assume this function will return stuff even for some QList::find() call.

How do I get the name of the container? I think I need to exclude multimap as well.

Reply:
that info is not available through the container pointer as far as I know.. I think you have to parse the variable declaration.